workflows_core/dataset/dataset.py

```diff
@@ -52,7 +53,10 @@ def insert_documents(
             if hasattr(document, "to_dict"):
                 documents[index] = document.to_dict()
         return self._api._bulk_insert(
-            dataset_id=self._dataset_id, documents=documents, *args, **kwargs
+            dataset_id=self._dataset_id,
+            documents=json_encoder(documents),
```

Comment:
Can you please add the option the make json encoder optional in case there's a breaking bug